packages/tangle/scheduler.go

```diff
@@ -331,14 +372,29 @@ func (s *Scheduler) schedule() *Message {
 		msg := q.Front()
 		// a message can be scheduled, if it is ready
 		// (its issuing time is not in the future and all of its parents are eligible).
-		if msg != nil && !clock.SyncedTime().Before(msg.IssuingTime()) {
-			// compute how often the deficit needs to be incremented until the message can be scheduled
-			remainingDeficit := math.Dim(float64(msg.Size()), s.getDeficit(q.NodeID()))
-			nodeMana := s.accessManaCache.GetCachedMana(q.NodeID())
-			// find the first node that will be allowed to schedule a message
-			if r := int(math.Ceil(remainingDeficit / nodeMana)); r < rounds {
-				rounds = r
-				schedulingNode = q
+		// while loop to skip all the confirmed messages
+		foundMsg := false
+		for msg != nil && !clock.SyncedTime().Before(msg.IssuingTime()) && !foundMsg {
+			msgID, _, err := MessageIDFromBytes(msg.IDBytes())
+			if err != nil {
+				panic("MessageID could not be parsed!")
+			}
+			if s.tangle.ConfirmationOracle.IsMessageConfirmed(msgID) && clock.Since(msg.IssuingTime()) > time.Minute {
+				// if a message is confirmed, and issued some time ago, don't schedule it and take the next one from the queue
+				// do we want to mark those messages somehow for debugging?
+				s.Events.MessageSkipped.Trigger(msgID)
```

Comment:
This can be very dangerous as this loop must be as fast as possible and by triggering events we could risk that the consumer of the event  is not going to be fast

Reply:
But I get that it's only done for debugging purposes

Reply:
That's true. However I want to have a metric with number of skipped messages in Grafana, and the handler of that event is just incrementing a number. Of course we need to remember in the future not to add any complex handlers there.

Reply:
I was more worried about the confirmation check, because the messages can be also removed upon confirmation, but doing it this way seemed easier to start with. Given that usually the queues should be relatively small, it might make sense to remove them upon confirmation. What do you think?